source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -913,6 +913,133 @@ void Graph::updateMaterials(mx::InputPtr input /* = nullptr */, mx::ValuePtr val
     }
 }
 
+// Set the node to display in render view based on selected node or nodegraph
+template <typename T>
+bool Graph::renderEnumInput(mx::InputPtr& input, const mx::UIProperties& ui, T &evalue)
+{
+    
+  // Eg. <input name="subsurface_color" type="color3" value="0.2,0.6,0.05" 
+  //    enum="PRESET1,PRESET2,PRESET3,PRESET4" enumvalues="0.2,0.6,0.05, 0.2,0.7,0.07, 0.2,0.6,0.05, 0.2,0.6,0.05" />
+  
+  //Cases
+  //1. ENUM_ATTRIBUTE attribute is empty (`enum`)
+  //2. ENUM_VALUES_ATTRIBUTE attribute can be empty, only if its a string type (`enumvalues`)
+  //3. `value` is not in enumvalues
+  //4. ui.enumerationValues and ui.enumeration are not populated for non-uniforms, 
+  //    hence use enumvalues attribute directly
+
+  std::string enumStr = input->getAttribute(mx::Input::ENUM_ATTRIBUTE);
+  std::string enumValueStr = input->getAttribute(mx::Input::ENUM_VALUES_ATTRIBUTE);
+
+  if (enumStr.empty())
+  {
+    return false;
+  }
+
+  const size_t INVALID_INDEX = std::numeric_limits<size_t>::max();
+  size_t index = INVALID_INDEX;
+  mx::ValuePtr val = input->getValue();
+  evalue = val->asA<T>();
+  std::string enumDelimiter = ",";
+
+  auto splitStringByGroup = [](const std::string& input, size_t groupSize) {
+    std::vector<std::string> result;
+    std::stringstream ss(input);
+    std::string token;
+    std::string group;
+    size_t count = 0;
+
+    while (std::getline(ss, token, ',')) {
+        if (!group.empty()) {
+            group += ",";
+        }
+        group += token;
+        count++;
+
+        if (count == groupSize) {
+            result.push_back(group);
+            group.clear();
+            count = 0;
+        }
+    }
+
+    return result;
+ };
+
+  // target enumStr result in eg. ["PRESET1", "PRESET2", "PRESET3", "PRESET4"]
+  // target vec3 enumValues result in eg. ["0.2,0.6,0.05", "0.2,0.7,0.07", "0.2,0.6,0.05", "0.2,0.6,0.05"]
+  std::vector<mx::ValuePtr> enumValues = ui.enumerationValues;
+  std::vector<std::string> enumVec = ui.enumeration;
+
+  //Handling edge cases where enumvalues attribute is empty
```

Comment:
I notice some minor inconsistencies in these comments, and it would be valuable to use the same spacing and capitalization for all of them.  Usually, we use a single space after the initial `//` marker, and the first letter of the first sentence is capitalized.